nsxt/resource_nsxt_policy_host_transport_node.go

```diff
@@ -0,0 +1,313 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	model2 "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"golang.org/x/exp/maps"
+)
+
+func resourceNsxtPolicyHostTransportNode() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyHostTransportNodeCreate,
+		Read:   resourceNsxtPolicyHostTransportNodeRead,
+		Update: resourceNsxtPolicyHostTransportNodeUpdate,
+		Delete: resourceNsxtPolicyHostTransportNodeDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyHostTransportNodeImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"site_path": {
+				Type:         schema.TypeString,
+				Description:  "Path to the site this Host Transport Node belongs to",
+				Optional:     true,
+				ForceNew:     true,
+				Default:      defaultInfraSitePath,
+				ValidateFunc: validatePolicyPath(),
+			},
+			"enforcement_point": {
+				Type:        schema.TypeString,
+				Description: "ID of the enforcement point this Host Transport Node belongs to",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"discovered_node_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "Discovered node id to create Host Transport Node",
+				ConflictsWith: []string{"node_deployment_info"},
+			},
+			"node_deployment_info": getFabricHostNodeSchema(),
+			// host_switch_spec
+			"standard_host_switch":      getStandardHostSwitchSchema(),
+			"preconfigured_host_switch": getPreconfiguredHostSwitchSchema(),
+		},
+	}
+}
+
+func getFabricHostNodeSchema() *schema.Schema {
+	elemSchema := map[string]*schema.Schema{
+		"fqdn": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Fully qualified domain name of the fabric node",
+		},
+		"ip_addresses": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Computed:    true,
+			Description: "IP Addresses of the Node, version 4 or 6",
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validateSingleIP(),
+			},
+		},
+	}
+	maps.Copy(elemSchema, getHostNodeSchemaAddlElements())
+	s := schema.Schema{
+		Type:     schema.TypeList,
```

Comment:
Should this be `ForceNew`? Is the just updated if IP address changes?

Reply:
No, IP address could be updated in place.